Gordon360/Services/NewsService.cs

```diff
@@ -153,9 +173,156 @@ public StudentNewsViewModel EditPosting(int newsID, StudentNews newData)
             newsItem.Subject = newData.Subject;
             newsItem.Body = newData.Body;
 
+            if (newData.Image != null)
+            {
+                UploadImage(newData.Image, newsID);
+            }
+
+            //If the image property is null, it means that either the user
+            //chose to remove the previous image or that there was no previous
+            //image (RemoveImage is designed to handle this).
+            else
+            {
+                RemoveImage(newsItem.SNID);
+                newsItem.Image = newData.Image;
+            }
+
             _unitOfWork.Save();
             
-            return (StudentNewsViewModel)newsItem;
+            return newsItem;
+        }
+
+        /// <summary>
+        /// Takes a filepath for an image, navigates to it, collects the raw data
+        /// of the file and converts it to base64 format. 
+        /// 
+        /// The base64 data will not include the first part of a base64 image ("data:image/...").
+        /// This is because this part is removed in every image before being submitted, and it
+        /// is readded in the frontend before being displayed.
+        /// 
+        /// This helper function does not perform any error checking; every place that calls it
+        /// checks that the path is not empty. Theoretically if it isn't empty it's certainly a valid path.
+        /// </summary>
+        /// <param name="imagePath">The path to the image</param>
+        /// <returns>The base64 content of the image</returns>
+        public string GetBase64ImageDataFromPath(string imagePath)
```

Comment:
It's worth considering making an `ImageUtils.cs` class to implement these on. The advantages are:

1. They can be seamlessly reused wherever else they're needed.
2. They would no longer clutter up this file with stuff that isn't strictly news-related.